### ISSUE TITLE
Fix Reach.addMissing on Windows

### ISSUE THREAD

**Issue by david-bouyssie** [scala-native/scala-native#2322]
Compiling the testSuite of https://github.com/david-bouyssie/xml4s on Windows currently crashes on:
```
[error] java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/Dev/scala-native/xml4s/xmlpull/shared/src/main/scala/org/xmlpull/v1/XmlPullParserFactory.scala
[error]         at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
[error]         at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
[error]         at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
[error]         at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
[error]         at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
[error]         at java.nio.file.Paths.get(Paths.java:84)
[error]         at scala.scalanative.linker.Reach.addMissing(Reach.scala:792)
[error]         at scala.scalanative.linker.Reach.$anonfun$processDelayed$2(Reach.scala:127)
[error]         at scala.scalanative.linker.Reach.$anonfun$processDelayed$2$adapted(Reach.scala:120)
[error]         at scala.Option.foreach(Option.scala:407)
[error]         at scala.scalanative.linker.Reach.$anonfun$processDelayed$1(Reach.scala:120)
[error]         at scala.scalanative.linker.Reach.$anonfun$processDelayed$1$adapted(Reach.scala:118)
[error]         at scala.collection.mutable.HashSet.foreach(HashSet.scala:79)
[error]         at scala.scalanative.linker.Reach.processDelayed(Reach.scala:118)
[error]         at scala.scalanative.linker.Reach$.apply(Reach.scala:822)
[error]         at scala.scalanative.linker.Link$.apply(Link.scala:12)
```

Stackoverflow related question:
https://stackoverflow.com/questions/43972777/exception-in-thread-main-java-nio-file-invalidpathexception-illegal-char

The solution might be to change the line of code:
https://github.com/scala-native/scala-native/blob/e855dda9606937bffebfc49723783d8f820acd50/tools/src/main/scala/scala/scalanative/linker/Reach.scala#L792

I have not tested but maybe this could work (if ```.getPath``` can be removed): 
```
val position = NonReachablePosition(path = Paths.get(pos.source),
```

**Comment by david-bouyssie** [comment]
@ekrich suggested to add a specific unit test for that kind of paths.

**Comment by david-bouyssie** [comment]
Note: the suggested solution (replacing "pos.source.getPath" by "pos.source") seems to work.

**Comment by WojciechMazur** [comment]
Already fixed and merged in #2344
